src/MongoDB.Bson/PreserveAttribute.cs

```diff
@@ -0,0 +1,38 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.ComponentModel;
+
+namespace MongoDB.Bson
+{
+    /// <summary>
+    /// Prevents the Xamarin managed linker from linking the target.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.All)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public sealed class PreserveAttribute : Attribute
```

Comment:
the xamarin version of this attribute is a bit different: https://github.com/johot/Xamarin.Forms.2.3.4/blob/d9421967dea90949f14048e9a44ea3092df5fffe/Xamarin.Forms.Core/Internals/PreserveAttribute.cs as well as it'a a bit different with doc: https://docs.microsoft.com/en-us/xamarin/ios/deploy-test/linker?tabs=windows#preserving-code. But, as I understand, it also works. So leaving it as suggested

Reply:
Can you look into the meaning of the following line and comment back?

    [EditorBrowsable(EditorBrowsableState.Never)]

Reply:
cc: @nirinchev

Reply:
Yes, the Xamarin linker appears to work with both fields and properties. I made these properties because the analyzer complained and didn't feel like suppressing warnings.

The `EditorBrowsable` attributes is a hint for the editor tooling (mainly intellisense) that it should not suggest the member it is applied on. The `PreserveAttribute` needs to be public for the linker to pick it up, but it is not part of the driver API and is unlikely to be useful for people. So if someone types `MongoDB.Bson.P`, they shouldn't get autocomplete suggestions for `PreserveAttribute`.

Reply:
Thanks for the info on `EditorBrowsable`. That seems like a good idea.

I suggest we keep the fields and suppress the warnings (suppress in this file only).

If we omit them someone looking at this file in the future might conclude we forgot them by mistake.

Reply:
>I suggest we keep the fields and suppress the warnings (suppress in this file only).

done